openmls/src/group/tests/test_commit_validation.rs

```diff
@@ -672,3 +672,75 @@ fn test_valsem205(ciphersuite: Ciphersuite, backend: &impl OpenMlsCryptoProvider
         .process_message(backend, MlsMessageIn::from(original_plaintext))
         .expect("Unexpected error.");
 }
+
+// this ensures that a member can process commits not containing all the stored proposals
+#[apply(ciphersuites_and_backends)]
+fn test_partial_proposal_commit(ciphersuite: Ciphersuite, backend: &impl OpenMlsCryptoProvider) {
+    // Test with PublicMessage
+    let CommitValidationTestSetup {
+        mut alice_group,
+        mut bob_group,
+        ..
+    } = validation_test_setup(PURE_PLAINTEXT_WIRE_FORMAT_POLICY, ciphersuite, backend);
+    let charlie_credential = generate_credential_bundle(
+        "Charlie2".into(),
+        CredentialType::Basic,
+        ciphersuite.signature_algorithm(),
+        backend,
+    )
+    .unwrap();
+    let dave_credential = generate_credential_bundle(
+        "Dave".into(),
+        CredentialType::Basic,
+        ciphersuite.signature_algorithm(),
+        backend,
+    )
+    .unwrap();
+
+    let charlie_key_package =
+        generate_key_package(&[ciphersuite], &charlie_credential, vec![], backend)
+            .expect("An unexpected error occurred.");
+    let dave_key_package = generate_key_package(&[ciphersuite], &dave_credential, vec![], backend)
+        .expect("An unexpected error occurred.");
+
+    let process_msg = |group: &mut MlsGroup, msg: MlsMessageOut| -> ProcessedMessageContent {
+        let serialized = msg.tls_serialize_detached().unwrap();
+        let plaintext = MlsMessageIn::tls_deserialize(&mut serialized.as_slice())
+            .unwrap()
+            .into_plaintext()
+            .unwrap();
+        group
+            .process_message(backend, MlsMessageIn::from(plaintext))
+            .unwrap()
+            .into_content()
+    };
+    let process_proposal = |group: &mut MlsGroup, msg: MlsMessageOut| {
+        let msg = process_msg(group, msg);
+        match msg {
+            ProcessedMessageContent::ProposalMessage(p) => group.store_pending_proposal(*p),
+            _ => panic!("Unexpected message type"),
+        }
+    };
+
+    // Create Proposals in Alice's group
+    let msg_out = alice_group
+        .propose_add_member(backend, &charlie_key_package)
+        .unwrap();
+    // Bob processes the proposal
+    process_proposal(&mut bob_group, msg_out);
+    let msg_out = alice_group
+        .propose_add_member(backend, &dave_key_package)
+        .unwrap();
+    // Bob processes the proposal
+    process_proposal(&mut bob_group, msg_out);
+
+    // Alice makes a partial commit
+
+    // Drop one proposal from the queue so the commit includes only one proposal
+    // NOTE: there's no API to make a commit including selected proposals
+    alice_group.proposal_store.pop().unwrap();
```

Comment:
I know it's painful but could you do the same without introducing this method ?

Reply:
can you check if this is better now?